sharing/src/test/scala/io/delta/sharing/spark/DeltaSharingDataSourceDeltaSuite.scala

```diff
@@ -0,0 +1,571 @@
+        DeltaSharingUtils.overrideIteratorBlock[String](
+          blockId = TestClientForDeltaFormatSharing.getBlockId(sharedTableName, "getMetadata"),
+          // scalastyle:off line.size.limit
+          values = Seq(
+            """{"add": {"path":"random","id":"random","partitionValues":{},"size":1,"motificationTime":1,"dataChange":false}}"""
+          ).toIterator
+        )
+        val exception2 = intercept[IllegalStateException] {
+          spark.read.format("deltaSharing").option("responseFormat", "delta").load(tablePath).schema
+        }
+        assert(
+          exception2.getMessage
+            .contains(s"Failed to get Protocol for $tableFullName")
+        )
+
+        // table_with_broken_metadata
+        // able to parse as a DeltaSharingSingleAction, but it's an addFile, not metadata.
+        DeltaSharingUtils.overrideIteratorBlock[String](
+          blockId = TestClientForDeltaFormatSharing.getBlockId(sharedTableName, "getMetadata"),
+          values = Seq(
+            """{"protocol":{"minReaderVersion":1}}"""
+          ).toIterator
+        )
+        val exception3 = intercept[IllegalStateException] {
+          spark.read.format("deltaSharing").option("responseFormat", "delta").load(tablePath).schema
+        }
+        assert(
+          exception3.getMessage
+            .contains(s"Failed to get Metadata for $tableFullName")
+        )
+      }
+
+      withSQLConf(getDeltaSharingClassesSQLConf.toSeq: _*) {
+        val profileFile = prepareProfileFile(tempDir)
+        val tableFullName = s"share1.default.$sharedTableName"
+        test(s"${profileFile.getCanonicalPath}#$tableFullName", tableFullName)
+      }
+    }
+  }
+
+  /**
+   * snapshot queries
+   */
+  test("DeltaSharingDataSource able to read simple data") {
+    withTempDir { tempDir =>
+      val deltaTableName = "data_source_suite_t"
+      withTable(deltaTableName) {
+        createTable(deltaTableName)
+        sql(
+          s"INSERT INTO $deltaTableName" +
+          """ VALUES (1, "one", "2023-01-01", "2023-01-01 00:00:00"),
+              |(2, "two", "2023-02-02", "2023-02-02 00:00:00")""".stripMargin
+        )
+
+        val sharedTableName = "shared_table"
+        prepareMockedClientAndFileSystemResult(deltaTableName, sharedTableName)
+        prepareMockedClientGetTableVersion(deltaTableName, sharedTableName)
+
+        val expectedSchema: StructType = new StructType()
+          .add("c1", IntegerType)
+          .add("c2", StringType)
+          .add("c3", DateType)
+          .add("c4", TimestampType)
+        val expected = Seq(
+          Row(1, "one", sqlDate("2023-01-01"), sqlTimestamp("2023-01-01 00:00:00")),
+          Row(2, "two", sqlDate("2023-02-02"), sqlTimestamp("2023-02-02 00:00:00"))
+        )
+
+        def test(tablePath: String): Unit = {
+          assert(
+            expectedSchema == spark.read
+              .format("deltaSharing")
+              .option("responseFormat", "delta")
```

Comment:
I see the code also handles the Parquet format. Can you add tests for the same?

Reply:
`DeltaSharingDataSource able to auto resolve responseFormat` tested it.

Reply:
and the parquet format sharing is tested in delta-sharing-client unit tests.